source/mir/random/engine/package.d

```diff
@@ -259,32 +259,71 @@ pragma(inline, true)
     return seed;
 }
 
+// Is llvm_readcyclecounter supported on this platform?
+// We need to whitelist platforms where it is known to work because if it
+// isn't supported it will compile but always return 0.
+// https://llvm.org/docs/LangRef.html#llvm-readcyclecounter-intrinsic
+version(LDC)
+{
+    // The only architectures the documentation says are supported are
+    // x86 and Alpha. x86 uses RDTSC and Alpha uses RPCC.
+    version(X86_64) version = LLVMReadCycleCounter;
+    // Do *not* support 32-bit x86 because some x86 processors don't
+    // support `rdtsc` and because on x86 (but not x86-64) Linux
+    // `prctl` can disable a process's ability to use `rdtsc`.
+    else version(Alpha) version = LLVMReadCycleCounter;
+}
+
+
 pragma(inline, false)
 private ulong fallbackSeed()()
 {
     // fallback to old time/thread-based implementation in case of errors
-    version(Windows)
+    version(D_InlineAsm_X86_64)
+    {
+        // RDTSC takes around 22 clock cycles.
+        ulong ticks = void;
+        asm @nogc nothrow
+        {
+            rdtsc;
+            shl RDX, 32;
+            xor RDX, RAX;
+            mov ticks, RDX;
+        }
+    }
+    //else version(D_InlineAsm_X86)
+    //{
+    //    // We don't use `rdtsc` with version(D_InlineAsm_X86) because
+    //    // some x86 processors don't support `rdtsc` and because on
+    //    // x86 (but not x86-64) Linux `prctl` can disable a process's
+    //    // ability to use `rdtsc`.
+    //    static assert(0);
+    //}
+    else version(LLVMReadCycleCounter)
```

Comment:
Any reason to not prefer LLVM first instead of `D_InlineAsm_X86_64`?

Reply:
Thinking about it, no not really. I'll move it first.